### PR TITLE
Feature/create test get-results service

### PR DESCRIPTION
**Pull request by tanyushanel**
Done without observables$ | async in templates cos have no idea how to use observables as matdatasource. Had tried, but didn't have a result.
Services for test are done as global considering to reuse them in other modules. 
There is a hard-coded userId in service just to show receiving results from server. Will be removed.
image

### CODE REVIEW

File: src/app/models/user-profile/user-profile.component.ts [hunk 1]
```diff
@@ -17,14 +19,28 @@ export class UserProfileComponent implements OnInit {
...
   get testsCount() {
     return this.results.length;
   }
 
-  ngOnInit() {
-    this.results = [...MOCK_TEST_RESULTS];
+  setTestResults(): void {
+    if (this.authStoreService.user) {
+      const userId = this.authStoreService.user.id;
+      this.testHttpService.getTestResults(userId).subscribe((tests: Test[]) => {
```
Comment (Inunor) [inline src/app/models/user-profile/user-profile.component.ts:40]: As I understand you would use this.results inside html file. You shouldn't make it that way. You can type results | async inside your html file, plus I think it is better to have testHttpStoreService as well. As we discussed -> httpService and storeService comes together

Comment (Inunor) [reply src/app/models/user-profile/user-profile.component.ts:40]: This setTestResults should be insde TestStoreHttpService

**Review by Inunor — CHANGES_REQUESTED** [review]

**Comment by Inunor** [comment]
Any progress or do you have a problem with this pr?

File: src/constants/route-constant.ts [hunk 2]
```diff
@@ -20,5 +20,7 @@ export enum SidebarMenu {
   statistics = 'Statistics',
 }
 
+export const BASE_API_URL = 'http://elevel-001-site1.btempurl.com';
```
Comment (RomPavel) [inline src/constants/route-constant.ts:23]: Please reuse this variable in LoginApiUrl and  RefreshUserDataUrl

File: src/constants/mock-test-results.ts [hunk 3]
```diff
@@ -1,52 +1,53 @@
...
-      },
-      speaking: {
-        id,
-        mark: generateRandomMark(),
-      },
-    },
-    feedback: feedback[Math.floor(Math.random() * 5)],
-  };
-}
+// export function createNewTestResult(id: number): Test {
```
Comment (RomPavel) [inline src/constants/mock-test-results.ts:23]: please remove all mocks if you dont need them

File: src/app/models/user-profile/user-profile.component.ts [hunk 4]
```diff
@@ -17,14 +18,20 @@ export class UserProfileComponent implements OnInit {
...
     return this.results.length;
   }
 
+  constructor(private router: Router, private testStoreService: TestStoreService) {}
+
   ngOnInit() {
-    this.results = [...MOCK_TEST_RESULTS];
+    // this.results$ = [...MOCK_TEST_RESULTS];
+
+    this.testStoreService.getTestResults().subscribe({
```
Comment (RomPavel) [inline src/app/models/user-profile/user-profile.component.ts:30]: It should be in another way, as @Inunor proposed.
You just call this.testStoreService.getTestResults() here, but in testStoreService you subscribe to result and store result in a Observable variable (fox example **testsResults$**). And in this component you use **testsResults$**

File: src/app/models/user-profile/user-profile.component.ts [hunk 5]
```diff
@@ -17,14 +18,20 @@ export class UserProfileComponent implements OnInit {
...
-
   get testsCount() {
     return this.results.length;
   }
 
+  constructor(private router: Router, private testStoreService: TestStoreService) {}
+
   ngOnInit() {
-    this.results = [...MOCK_TEST_RESULTS];
+    // this.results$ = [...MOCK_TEST_RESULTS];
```
Comment (RomPavel) [inline src/app/models/user-profile/user-profile.component.ts:28]: Please remove all the comments

File: src/app/models/user-profile/user-profile.component.ts (hunk 4 above)
Comment (tanyushanel) [reply src/app/models/user-profile/user-profile.component.ts:30]: I tried? but have no idea how to use observable as matdatasourse.

**Comment by RomPavel** [comment]
> Done without observables$ | async in templates cos have no idea how to use observables as matdatasource. Had tried, but didn't have a result.
> Services for test are done as global considering to reuse them in other modules.
> There is a hard-coded userId in service just to show receiving results from server. Will be removed.


Did you try this?
image

File: src/app/models/user-profile/user-profile.component.ts (hunk 4 above)
Comment (RomPavel) [reply src/app/models/user-profile/user-profile.component.ts:30]: > > Done without observables$ | async in templates cos have no idea how to use observables as matdatasource. Had tried, but didn't have a result.
> > Services for test are done as global considering to reuse them in other modules.
> > There is a hard-coded userId in service just to show receiving results from server. Will be removed.
> 
> Did you try this?
> image

Comment (tanyushanel) [reply src/app/models/user-profile/user-profile.component.ts]: image
Seems like a magic-stick. Zzzz...

File: src/app/models/user-profile/user-profile.component.ts [hunk 6]
```diff
@@ -1,30 +1,28 @@
...
-  constructor(private router: Router) {}
-
-  get testsCount() {
-    return this.results.length;
-  }
+  constructor(private router: Router, private testStoreService: TestStoreService) {}
 
   ngOnInit() {
-    this.results = [...MOCK_TEST_RESULTS];
+    this.results$ = this.testStoreService.getTestResults();
```
Comment (RomPavel) [inline src/app/models/user-profile/user-profile.component.ts:25]: its still not in a proper way.
getTestResults() should not really return anything.
you should have testResults$ variable in your store service and getTestResults should set new value of  testResults$

File: src/app/services/test/test/test-store.service.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  private set testResults(testResults: Test[]) {
+    this.testSubject$.next(testResults);
+  }
+
+  constructor(
+    private testHttpService: TestHttpService,
+    private authStoreService: AuthStoreService,
+  ) {}
+
+  setTestResults(): void {
```
Comment (RomPavel) [inline src/app/services/test/test/test-store.service.ts:25]: I would stick with getTestResults. Meaning of get in naming is not like 'get result of this function' but more like 'get tests results from backend'

**Review by RomPavel — APPROVED** [review]

**Review by Inunor — APPROVED** [review]

**Pull request merged**